app/helpers/application_helper.rb

```diff
@@ -727,7 +727,7 @@ def setup_map_tag_attrs(options = {})
       "taxon-range-layer-label" => I18n.t("maps.overlays.range"),
       "taxon-places-layer-label" => I18n.t("maps.overlays.checklist_places"),
       "taxon-places-layer-hover" => I18n.t("maps.overlays.checklist_places_description"),
-      "taxon-observations-layer-label" => I18n.t("maps.overlays.observations"),
+      "taxon-observations-layer-label" => I18n.t( options[:taxon_observations_layer_label] ? options[:taxon_observations_layer_label] : "maps.overlays.observations" ),
```

Comment:
Can you make the entire content of the label optional instead of the translation key? Like

```ruby
"taxon-observations-layer-label" => options[:taxon_observations_layer_label].blank? ? I18n.t( "maps.overlays.observations" ) : options[:taxon_observations_layer_label],
```

Having stuff other than a string literal inside the `t( )` makes it much harder to analyze the code for things like unused translations.